### PR TITLE
Gauge with decimal digits

### PR DESCRIPTION
**Pull request by fabian-m-95**
## Hey, I just made a Pull Request!

I added the functionality to display decimal values to the Gauge plugin, as described in https://github.com/backstage/backstage/issues/25026

Example with 5.5/10:
gauge-decimals

To achieve this, I added to new properties:
* `relativeToMax`: properly fills and colors the circle, even if the maximum is not 100
* `decimalDigits`: indicates to how many digits the displayed number should be rounded

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/api-report.md [hunk 1]
```diff
@@ -450,6 +450,8 @@ export type GaugeProps = {
   size?: 'normal' | 'small';
   description?: ReactNode;
   getColor?: GaugePropsGetColor;
+  relativeToMax?: boolean;
```
Comment (freben) [inline packages/core-components/api-report.md:453]: Not sure about this one. If a max was given, when would you NOT want the gauge to be relative to the max? What are the behaviors when this is true vs false? My gut reaction is, this should always be true - right?

Comment (fabian-m-95) [reply packages/core-components/api-report.md:453]: Thank you for your feedback. I generally agree with you, and that is also what I would expect. However, there is one test which explicitly renders (value = 1, max = 10) as "1%": https://github.com/backstage/backstage/blob/master/packages/core-components/src/components/ProgressBars/Gauge.test.tsx#L40
The intention behind that test is not clear to me, so I opted not to change the behavior.

I would gladly tidy up more code, but without knowing why the code was written the way it is I didn't dare to. Please note that such a change also wouldn't be backwards compatible.

Comment (fabian-m-95) [reply packages/core-components/api-report.md:453]: Hi @freben, have you already had the opportunity to have a closer look?

In which way should I proceed? Keep by merge request as-is or make "relativeToMax" the default (if max exists), possibly breaking existing applications?

**Review by freben — APPROVED** [review]
Mmmm okay. I went back and forth and wasn't sure where to best go with this - we're in a kinda odd position where the component does "too much" and there are some combinations of props that don't work well etc.  But sure, this was a feature / ability that was missing, and I don't have any great suggestions for what to do about it.

**Pull request merged**